portfolio/src/main/java/com/google/sps/servlets/RequestServlet.java

```diff
@@ -35,10 +48,15 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
             return;
         }
 
+        List<Object> fetchable_items = new ArrayList<Object>();
+        fetchable_items.add(results);
+        fetchable_items.add(authenticationUrl);
+        fetchable_items.add(loggedin);
+
```

Comment:
We're creating a heterogeneous JSON list here. It would be better to create a JSON map, in which these three items have names. That would also make the javascript code easier to understand, because we'd have `if (tasks["isLoggedIn"])` instead of `if (tasks[0] == "true")`, and other similar improvements.

I'm not totally sure how to do this, but I think I've seen you (or maybe one of the others) do this with comments, so I know it's possible. Ask around.